### PR TITLE
Build fixes + comments

### PR DESCRIPTION
**Pull request by erikandre**

### CODE REVIEW

File: app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java [hunk 1]
```diff
@@ -21,8 +18,9 @@
 
 public final class ExceptionReporter {
 
-    private static ExceptionReporter instance;
-    private Repository repo;
+    //ERIK: Pls follow the Google naming convention: https://source.android.com/source/code-style
```
Comment (testneer) [inline app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java:21]: yeah I have it in my todo. Thanks

File: app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java [hunk 2]
```diff
@@ -36,6 +34,8 @@ private ExceptionReporter(Context ctx){
 
     };
 
+    //ERIK - is there any specific reason why you require this to be an Application and not just a Context?
```
Comment (testneer) [inline app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java:37]: yeah i thought about it, wasn't sure if it's any better to call context.getApplicaitonContext and left it as is

File: app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java [hunk 3]
```diff
@@ -48,11 +48,12 @@ public static ExceptionReporter getInstance(Application app){
 
 
     public void init(){
+        //ERIK - it might be nice to prevent this from being called twice
         interceptExceptions();
     }
 
     private void interceptExceptions() {
-        Thread.UncaughtExceptionHandler defaultExceptionHandler = Thread.getDefaultUncaughtExceptionHandler();
+        final Thread.UncaughtExceptionHandler defaultExceptionHandler = Thread.getDefaultUncaughtExceptionHandler();
```
Comment (testneer) [inline app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java:56]: why does it have to be final?

File: app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java [hunk 4]
```diff
@@ -72,6 +73,7 @@ public static void logException(Exception caughtException){
     }
 
     //TODO: Not thread safe
+    //ERIK - why not thread save? I think only one uncaught exception handler can execute at one time anyways
```
Comment (testneer) [inline app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java:76]: well.. this is for caught execptions.. imagine i spin another thread in my app and then call on this method.. not sure how it'll handle things

File: app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/data/DiskStore.java [hunk 5]
```diff
@@ -68,8 +53,8 @@ public void saveReportDataToFile(@NonNull ExceptionReport exceptionReport) throw
         }
     }
 
-    public File[] getUnsentReport() {
-        File dir = ctx.getFilesDir();
+    public File[] getUnsentReport() { //ERIK getUnsentReport(s)?
+        File dir = ctx.getFilesDir(); //ERIK - you definitely want to create and use a subdirectory of the files folder. To avoid sending and deleting other files
```
Comment (testneer) [inline app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/data/DiskStore.java:58]: :)))

**Review by testneer — APPROVED** [review]

File: app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java (hunk 3 above)
Comment (erikandre) [reply app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java:56]: In Android Studio 3.0, Google does some magic to allow you to use Java 8 language features, like automatically making stuff static when referencing them from inside an anonymous inner class. To make it build in AS 2.3 I dropped Java 8 support (you could still get that using RetroLambda if needed)

File: app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java (hunk 2 above)
Comment (erikandre) [reply app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java:37]: Yeah, I think just using context is always better unless you got a very specific requirement.

File: app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java (hunk 4 above)
Comment (erikandre) [reply app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java:76]: Ah, I see what you mean now. Just make the method synchronized then. Performance is not going to be a factor for this kind of code (I mean, if you enough logged exceptions to cause a slowdown, then something else is probably wrong, haha)

Comment (testneer) [reply app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java:76]: makes sense

File: app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java (hunk 3 above)
Comment (testneer) [reply app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java:56]: got it.. yeah.. it'll be a lib so it's better to drop support for it. cheers

File: app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java (hunk 2 above)
Comment (testneer) [reply app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java:37]: 👍

File: app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java (hunk 4 above)
Comment (testneer) [reply app/src/main/java/com/ex/org/exceptionstest/com/ex/lib/ExceptionReporter.java:76]: hmm.. actually i don't think that making it block is a good idea... I will leave a comment to add this e to a blockingqueue and release the caller. These might be debug exceptions coming from multiple threads

**Pull request merged**